src/lib.rs

```diff
@@ -751,6 +846,34 @@ pub unsafe trait FromZeroes {
     /// Self::new_zeroed()`, it differs in that `zero` does not semantically
     /// drop the current value and replace it with a new one - it simply
     /// modifies the bytes of the existing value.
+    ///
+    /// ```
+    /// # mod derive { pub use zerocopy_derive::FromZeroes; }
+    /// # use zerocopy::FromZeroes;
+    /// # use derive::*;
+    /// #
+    /// # #[derive(FromZeroes)]
+    /// # #[repr(C)]
+    /// # struct PacketHeader {
+    /// #     src_port: [u8; 2],
+    /// #     dst_port: [u8; 2],
+    /// #     length: [u8; 2],
+    /// #     checksum: [u8; 2],
+    /// # }
+    /// let mut header = PacketHeader {
+    ///     src_port: 100u16.to_be_bytes(),
+    ///     dst_port: 200u16.to_be_bytes(),
+    ///     length: 300u16.to_be_bytes(),
+    ///     checksum: 400u16.to_be_bytes(),
+    /// };
+    ///
+    /// header.zero();
+    ///
+    /// assert_eq!([0, 0], header.src_port);
+    /// assert_eq!([0, 0], header.dst_port);
+    /// assert_eq!([0, 0], header.length);
+    /// assert_eq!([0, 0], header.checksum);
```

Comment:
To be consistent with [stdlib examples](https://doc.rust-lang.org/stable/std/ptr/struct.NonNull.html#examples-3), swap the order here:

```suggestion
    /// assert_eq!(header.src_port, [0, 0]);
    /// assert_eq!(header.dst_port, [0, 0]);
    /// assert_eq!(header.length, [0, 0]);
    /// assert_eq!(header.checksum, [0, 0]);
```